src/dotnet-interactive/Program.cs

```diff
@@ -99,30 +99,31 @@ internal static IDisposable StartToolLogging(StartupOptions options)
 
             return webHost;
 
-            static HttpPort GetFreePort(StartupOptions startupOptions)
+            static HttpPort GetFreePort(StartupOptions startupOptions, int maxRetries = 50)
             {
                 using var __ = Log.OnEnterAndExit(nameof(GetFreePort));
                 if (startupOptions.HttpPort is not null && !startupOptions.HttpPort.IsAuto)
                 {
                     return startupOptions.HttpPort;
                 }
 
-                var currentPort = 0;
-                var endPort = 0;
+                var startPort = 49152; // RFC 6335 compatible
```

Comment:
0 values as default will let the  TCPListener pick a random one when not specified, this way is the HttpPortRance is not specified we keep scanning instead of just letting pick a random one